ql/src/java/org/apache/hadoop/hive/ql/io/parquet/read/DataWritableReadSupport.java

```diff
@@ -209,6 +257,57 @@ private static MessageType getSchemaByIndex(MessageType schema, List<String> col
   }
 
   /**
+   * Return the columns which contains required nested attribute level
+   * e.g.
+   * Given struct a <x:int, y:int> and a is required while y is not, so the method will return a
+   * who contains the attribution x
+   *
+   * @param groupPaths the paths for required nested attributions
+   * @return column list contains required nested attribution
+   */
+  private static Set<String> getPrunedNestedColumns(List<String> groupPaths) {
+    Set<String> colNames = new HashSet<>();
+    if (groupPaths.isEmpty()) {
+      return colNames;
+    }
+    for (String s : groupPaths) {
+      String c = StringUtils.split(s, '.')[0];
+      if (!colNames.contains(c)) {
+        colNames.add(c);
+      }
+    }
+    return colNames;
+  }
+
+  private static GroupType buildProjectedGroupType(
+    GroupType originalType,
+    List<Type> types) {
+    return new GroupType(originalType.getRepetition(), originalType.getName(), types);
```

Comment:
This may produce wrong results since it will mess up the index. For instance:
if I do `select s.c from tbl` where `tbl` is of schema `a int, s struct<b:int, c:string>`, then the
index for `c` is changed from original 1 to 0 in the new group type.